### PR TITLE
Update md files

### PR DESCRIPTION
**Pull request by GrayJack**
Not complete, lack some information for me

`with` form is a special form?
Can `match` form pattern match primitive types?

### CODE REVIEW

File: docs/LanguageGuide.md [hunk 1]
```diff
@@ -217,6 +219,23 @@ Sometimes, it's more convenient to bring a module's declarations into scope only
 (Vector2.update-x my-pos inc) ;; => (Vector2 11 20)
 ```
 
+### Enums / Algebraic types
+```clojure
+(deftype (Either a b)
+  (Left a)
```
Comment (hellerve) [inline docs/LanguageGuide.md]: The members here must be wrapped in array, i.e. `(Left [a])`.

File: docs/LanguageGuide.md [hunk 2]
```diff
@@ -75,6 +76,7 @@ and other static analysis. The first three of them are also available in dynamic
 (do <expr1> <expr2> ... <return-expression>) ;; Perform side-effecting functions, then return a value
 (if <expression> <true-branch> <false-branch>) ;; Branching
 (while <expression> <body>) ;; Loop until expression is false
+(match <expression> <case1> <expression1> <case2> <expression2> ...) ;; Pattern match the first expression in a exhaustive cases
```
Comment (hellerve) [inline docs/LanguageGuide.md]: `match` only works on sumtypes for now.

File: docs/LanguageGuide.md [hunk 3]
```diff
@@ -217,6 +219,23 @@ Sometimes, it's more convenient to bring a module's declarations into scope only
 (Vector2.update-x my-pos inc) ;; => (Vector2 11 20)
 ```
 
+### Enums / Algebraic types
```
Comment (hellerve) [inline docs/LanguageGuide.md]: We usually prefer the term sumtypes over algebraic types in Carp speak.

File: docs/LanguageGuide.md [hunk 4]
```diff
@@ -217,6 +219,23 @@ Sometimes, it's more convenient to bring a module's declarations into scope only
...
+```clojure
+(deftype (Either a b)
+  (Left a)
+  (Right b))
+
+;; A Variant can be created with the same syntax as call expression
+(Left 10)
```
Comment (hellerve) [inline docs/LanguageGuide.md]: You either need to `use` `Either` before you can do that, or you’ll have to write `Either.Left`.

File: docs/LanguageGuide.md (hunk 1 above)
Comment (GrayJack) [reply docs/LanguageGuide.md]: True, forgot about that when I wrote that

File: docs/LanguageGuide.md (hunk 2 above)
Comment (GrayJack) [reply docs/LanguageGuide.md]: It would be better to just add a note or change de entire comment?

Comment (hellerve) [reply docs/LanguageGuide.md]: Good question. How about `Pattern matches <expression> against a set of sumtype constructors`? I don’t know whether that is clear enough, though.

Comment (GrayJack) [reply docs/LanguageGuide.md]: There is plans to support pattern match on non sumtypes?

> I don’t know whether that is clear enough, though.

It's clear enought for me

Comment (hellerve) [reply docs/LanguageGuide.md]: Eventually yes, but we want to sort out a bunch of other fundamental things first. Contributions are always welcome, though! (Though this is likely to be a little too hard to add for someone who doesn’t have a deep understanding of the Carp compiler)

**Comment by GrayJack** [comment]
Another question pop up by looking at the stdlib source code,
In core/Color.carp has shown this pattern
```closure
(deftype Id
     Black
     Red
     Green
     Yellow
     Blue
     ...)
```

That means that sumtypes variants doesn't need to be between parenthesis when doesn't depend on anything?

I ask cause the pattern used by `Maybe`, when declared the variant `Nothing` put it between parenthesis
```closure
(deftype (Maybe a)
  (Nothing)
  (Just [a]))
```

**Comment by hellerve** [comment]
I stumbled upon the same thing within #621. Apparently this does work! I just didn’t remember.

**Comment by eriksvedang** [comment]
Yeah, perhaps we should have an example of that in the document too?

**Comment by eriksvedang** [comment]
This is awesome btw, thanks a lot @GrayJack !!!

File: docs/LanguageGuide.md [hunk 5]
```diff
@@ -220,12 +220,26 @@ Sometimes, it's more convenient to bring a module's declarations into scope only
...
+
+**Enumeration:**
+```clojure
+(deftype MyEnum
+  Kind1
+  Kind2
+  Kind3)
+```
+
+**Data:**
```
Comment (GrayJack) [inline docs/LanguageGuide.md]: I was not sure what to put in here, any suggestions?

File: docs/Tooling.md [hunk 6]
```diff
@@ -27,6 +27,9 @@ For Flycheck:
 
 To start an interactive session, make sure `carp` is in your path (inside Emacs) and execute `M-x run-carp`.
 
+# Atom
+[language-carp](https://atom.io/packages/language-carp) offers both TextMate and Tree-sitter, where tree-siiter grammar is more powerful
```
Comment (jacereda) [inline docs/Tooling.md]: There's a typo there (siiter)

Comment (GrayJack) [reply docs/Tooling.md]: There is no typo in 'sitter', pro project call tree-sitter, look out their site https://tree-sitter.github.io/tree-sitter/

Comment (jacereda) [reply docs/Tooling.md]: There is, continue reading after the comma.

Comment (GrayJack) [reply docs/Tooling.md]: Yeah, sorry, misted that, I was without glasses xD
Was hard to spot on

File: docs/Tooling.md [hunk 7]
```diff
@@ -27,6 +27,9 @@ For Flycheck:
 
 To start an interactive session, make sure `carp` is in your path (inside Emacs) and execute `M-x run-carp`.
 
+# Atom
+[language-carp](https://atom.io/packages/language-carp) offers highlight with both TextMate and Tree-sitter, where tree-sitter grammar is more powerful
```
Comment (hellerve) [inline docs/Tooling.md]: Is there a period missing at the end?

**Comment by eriksvedang** [comment]
Thanks a lot!

**Pull request merged**